bin/main

```diff
@@ -1,70 +1,121 @@
 #!/usr/bin/env ruby
+
 # rubocop: disable Style/StringLiterals
 
-# Welcome the player to the game
+# rubocop: disable Style/GuardClause
+
+# rubocop:disable Lint/UnreachableCode
+
+# rubocop: disable Style/ConditionalAssignment
+
+# rubocop: disable Style/MultipleComparison
+
+require_relative '../lib/board.rb'
+require_relative '../lib/player.rb'
+
+def get_choice(arg)
+  loop do
+    puts 'Please enter a number between 1 to 9: '
+    choice = gets.chomp.to_i
+    index = choice - 1
+    if arg.include? choice
+      return index
+      break
+    else
+      puts 'Wrong position, please choose another one'
+    end
+  end
+end
+
+def valid_move(arr, player_choice, player)
+  valid = false
+  until valid
+    if arr[player_choice] == 'X' || arr[player_choice] == 'O'
+      puts 'Wrong postion, please choose another one'
+      return valid
+    else
+      arr[player_choice] = player.token
+      valid = true
+      break
+    end
+  end
+end
```

Comment:
Must be split in 2: The loop must stay inside the main.rb but the validation must be a method from the `Board` class.
```rb
board.valid_move(player_choice, player)
```

Where you have
```rb
# Board

def valid_move(player_choice, player)
   if this.array[player_choice] == 'X' || arr[player_choice] == 'O'
      return false
    else
      return true

     # The next part can be implemented as another method 
     # update_board()
      arr[player_choice] = player.token
      valid = true # No longer needed
    end
end
```

Reply:
:100: